plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/simulate/SimulatedItemResponse.java

```diff
@@ -0,0 +1,171 @@
+            String id = in.readString();
+            Map<String, Object> doc = in.readMap();
+            this.data = new Data(index, type, id, doc);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeBoolean(isFailed());
+        out.writeBoolean(isVerbose());
+
+        if (isFailed()) {
+            out.writeThrowable(failure);
+        } else if (isVerbose()) {
+            out.writeVInt(processedDataList.size());
+            for (ProcessedData p : processedDataList) {
+                p.writeTo(out);
+            }
+        } else {
+            out.writeString(data.getIndex());
+            out.writeString(data.getType());
+            out.writeString(data.getId());
+            out.writeMap(data.getDocument());
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(Fields.ERROR, isFailed());
+        builder.field(Fields.VERBOSE, isVerbose());
+        if (isFailed()) {
+            builder.field(Fields.FAILURE, failure.toString());
+        } else if (isVerbose()) {
+            builder.startArray(Fields.PROCESSOR_STEPS);
+            for (ProcessedData processedData : processedDataList) {
+                builder.value(processedData);
+            }
+            builder.endArray();
+        } else {
+            builder.field(Fields.MODIFIED, data.isModified());
+            builder.field(Fields.DOCUMENT, data.getDocument());
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public RestStatus status() {
+        if (isFailed()) {
+            return RestStatus.BAD_REQUEST;
+        } else {
+            return RestStatus.OK;
+        }
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) { return true; }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        SimulatedItemResponse other = (SimulatedItemResponse) obj;
+        return Objects.equals(data, other.data) && Objects.equals(processedDataList, other.processedDataList) && Objects.equals(failure, other.failure);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(data, processedDataList, failure);
+    }
+
+    static final class Fields {
+        static final XContentBuilderString DOCUMENT = new XContentBuilderString("doc");
+        static final XContentBuilderString ERROR = new XContentBuilderString("error");
+        static final XContentBuilderString VERBOSE = new XContentBuilderString("verbose");
+        static final XContentBuilderString FAILURE = new XContentBuilderString("failure");
+        static final XContentBuilderString MODIFIED = new XContentBuilderString("modified");
+        static final XContentBuilderString PROCESSOR_STEPS = new XContentBuilderString("processor_steps");
```

Comment:
maybe just call is `processor_result`? After all it is the result of a single processor.